### PR TITLE
Coin

### PR DESCRIPTION
**Pull request by Meghag5**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

(Write your answer here.)

### Description

(Write your answer here.)
Coin Change Problem Maximum Number of ways
This problem is solved using dp.
Given a value N, if we want to make change for N cents, 
and we have infinite supply of each of S = { S1, S2, .. , Sm} valued coins,
 how many ways can we make the change? The order of coins doesn’t matter.
In this problem we have to count whether the values of subset of array is equal to sum or not.
if equal then we will return true else we will return false.
Input -
Enter the value of n andd m:
10 4
2 5 3 6
 Enter the value of array:
Output - 5

### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [x] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

(Write your answer here.)
#5445

### CODE REVIEW

**Comment by Meghag5** [comment]
@HarshCasper @ankitaggarwal23 pls review

File: C-Plus-Plus/dp/Maximum_ways_Coin_change_problem.cpp [hunk 1]
```diff
@@ -0,0 +1,75 @@
+/*
+Coin Change Problem Maximum Number of ways
+This problem is solved using dp.
+Given a value N, if we want to make change for N cents, 
+and we have infinite supply of each of S = { S1, S2, .. , Sm} valued coins,
+ how many ways can we make the change? The order of coins doesn’t matter.
+In this problem we have to count whether the values of subset of array is equal to sum or not.
+if equal then we will return true else we will return false.
+*/
```
Comment (HarshCasper) [inline C-Plus-Plus/dp/Maximum_ways_Coin_change_problem.cpp]: Use a more standard description

File: C-Plus-Plus/dp/Maximum_ways_Coin_change_problem.cpp [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+	cin>>n>>m;
+	int arr[m];
+    cout << "Enter the value of array: \n";
+	for(int i=0;i<m;i++){
+		cin>>arr[i];
+    }
+   
+    // count by recalling function
+	cout<<count(arr,m,n)<<endl;
+    return 0;
```
Comment (HarshCasper) [inline C-Plus-Plus/dp/Maximum_ways_Coin_change_problem.cpp]: Improper indentation

**Review by HarshCasper — CHANGES_REQUESTED** [review]

**Comment by Meghag5** [comment]
@HarshCasper pr updated pls review

**Comment by Meghag5** [comment]
@HarshCasper pr updated pls review

**Review by HarshCasper — APPROVED** [review]

**Pull request merged**